### DISCUSSION TITLE
Websocket Behaviour

### DISCUSSION THREAD

**Discussion by rehatkathuria** [oven-sh/bun#5907] [Q&A]
I have the following use case: 

1. User fetches a signed url from the server that allows them to upload directly to an external provider.
2. External provider notifies the server in the form of a webhook
3. Server needs to notify the user that the upload has finished processing

Initially, I considered using websockets to notify the client of this event but upon further reading it seems like the `publish` methods drop the message when sending to the original subscriber. Am I incorrect in presuming this would have been the best way to notify the client? Are there other websocket methods? 

I presumed I'd be able to simply do the following:

1. Client subscribes to identifier `foo`
2. Server receives notification from S3 about url that has finished processing
3. Server publishes a message to the websocket to all clients subscribed to `foo` (in this case a single client who is interested in events associated with their account)

Note: the `foo` here is for conveying the problem and realistically I would probably have it represent the user identifier

Any insight and help would be apprecaited!

**Comment by melroy89** [comment]
For all the API methods see the official docs: https://bun.sh/docs/api/websockets

**Reply by rehatkathuria** [reply]
Yep. I've gone through the documentation a few times but it looks like the publish methods don't send to the connecting client?

Can you tell me if there's a way to isolate the websocket connection per client and only message them?

I have attached my current implementation in the following gist:

https://gist.github.com/rehatkathuria/dc7050101aefcae2b9e716d741885d27
